src/components/ICheckbox/ICheckbox.vue

```diff
@@ -0,0 +1,216 @@
+     * @name modelValue
+     */
+    modelValue: {
+        default: false
+    },
+    /**
+     * The unique identifier of the checkbox
+     * @type String
+     * @default uid()
+     * @name name
+     */
+    name: {
+        type: String,
+        default() {
+            return uid('checkbox');
+        }
+    },
+    /**
+     * Displays the native browser checkbox input indicator
+     * @type Boolean
+     * @default false
+     * @name native
+     */
+    native: {
+        type: Boolean,
+        default: false
+    },
+    /**
+     * The readonly state of the checkbox
+     * @type Boolean
+     * @default false
+     * @name readonly
+     */
+    readonly: {
+        type: Boolean,
+        default: false
+    },
+    /**
+     * The size variant of the checkbox
+     * @type sm | md | lg
+     * @default md
+     * @name size
+     *
+     */
+    size: {
+        type: String,
+        default: undefined
+    },
+    /**
+     * The tabindex of the checkbox
+     * @type Number | String
+     * @default 0
+     * @name tabindex
+     */
+    tabindex: {
+        type: [Number, String],
+        default: 0
+    }
+});
+
+const emit = defineEmits([
+    /**
+     * Event emitted for setting the modelValue
+     * @event update:modelValue
+     */
+    'update:modelValue'
+]);
+
+const input = ref<HTMLInputElement | null>(null);
+
+const checkboxGroup = inject(CheckboxGroupKey, {});
+
+const componentColor = useComponentColor({ componentName, currentColor: props.color });
+const componentSize = useComponentSize({ componentName, currentSize: props.size });
+const { disabled, size, readonly } = useInputState({
+    disabled: props.disabled,
+    readonly: props.readonly,
+    size: componentSize.value
+});
+const { schema, onInput, onBlur } = useValidation({
+    name: props.name,
+    elementType: 'checkbox'
+});
+
+const classes = computed(() => ({
+    [`-${componentColor.value}`]: true,
+    [`-${size.value}`]: true,
+    '-disabled': disabled.value,
+    '-readonly': readonly.value,
+    '-native': props.native
+}));
+
+const checked = computed(() => {
+    if (schema.value) {
+        return schema.value.value;
+    } else if (checkboxGroup?.values) {
+        return checkboxGroup.values.includes(props.value);
+    }
+    return props.modelValue;
+});
+
+const tabindex = computed(() => {
+    return disabled ? -1 : props.tabindex;
+});
+
+function onClickLabel() {
+    if (readonly.value) {
+        return;
+    }
+    input.value?.click();
+}
+
+function changed(event: Event) {
+    const target = event.target as HTMLInputElement;
+
+    if (checkboxGroup.onChange) {
+        checkboxGroup.onChange(props.value);
+    } else {
+        onInput(props.name, target.checked);
+    }
+    emit('update:modelValue', target.checked);
+}
+
+function blurred(event: FocusEvent) {
+    if (checkboxGroup.onBlur) {
+        checkboxGroup.onBlur(props.name, event);
+    } else {
+        onBlur(props.name, event);
+    }
+}
```

Comment:
nitpick: We're using `on[EventName]` in the rest of the components. Would love to keep consistent and use `onClick` and `onBlur` for naming.